### ISSUE TITLE
I can't see the draft helper

### ISSUE THREAD

**Issue by pablolp** [Manuel-777/MTG-Arena-Tool#70]
I don't know if it's an issue but I Can't see the draft helper... how can I open it? I hace the Version 2.1.10 already
![image](https://user-images.githubusercontent.com/12608340/47108637-76f64a80-d24c-11e8-976f-79f24bd5f166.png)

**Comment by Manuel-777** [comment]
You can force it open with overlay->persistent overlay in the settings page.
If that doesn't work try restarting mtgatool.

**Comment by pablolp** [comment]
I've also tried that and now I can see the overlay but still not working for the draft

**Comment by andybergon** [comment]
I also had some trouble having it to show up. I tried to do a fresh install but nothing. I'll try to clear all files related to the old installation and do another reinstall, hoping that it will fix it.

Probably makes sense to dig into it and make it more reliable.

**Comment by farazs** [comment]
@Manuel-777 I was debugging the app and it looks like there are several issues with the draft helper.

1. One is if the Tool starts up when Arena is running, the logs are parsed before the database with all the card data is received so the draft doesn't match any sets. The parsing should probably wait on the database fetch promise. A workaround is to exit out and re-enter the draft in Arena after the Tool is loaded so this isn't a big deal.
2. The draft is now for Dominaria. The eventName in Arena is `QuickDraft_DOM_10262018`. The Tool checks this against the set codes in the database and for dominaria this is `DAR` not `DOM`. Also, I noticed the database json is now fetched from your server rather than locally. I assume this is so you can add cards without doing an app release. I downloaded the json file, changed the code to use the file and updated the code for Dominaria to get around this
3. In background.js you check `httpAsync.length == 0` before `ipc.send("set_draft_cards"` and the length seems to always be 6 so the call is never made. I removed this check to get around it.
4. get_picks seems to return an empty response

So even after bypassing 2 and 3, the cards now show up with all of them ranked at 10 because the pick ranking returns empty.

**Comment by Manuel-777** [comment]
@farazs please check out latest release and commit [fd56ae](https://github.com/Manuel-777/MTG-Arena-Tool/commit/fd56ae104d83a3e22c339dd456a34ba2a40a7e2e), I removed the draft_picks cal and added picks ranking to db.json so I can update manually.

I am also starting to provide the most up-to-date db.json with each update, so the diff between server and default database shouldnt be too large, probably changing events will be affected from time to time.

On point 1, this is depending on the network speed. I do not want to restrict the user to be able to log in if the network is still loading the database, but given the new changes this should have minimal impact now.

About point 3, [just removed it](https://github.com/Manuel-777/MTG-Arena-Tool/commit/f91c0adf370d477600930815a5290506a947c58e), it was a dirty hack and is no longer needed since the ranks are provided by db from now on.

**Comment by farazs** [comment]
Removing this check `if (json.packNumber == 0 && json.pickNumber <= 0) {` from the draft code let's it work for me. I am in a draft where I am already on pick 3. Is there a reason to only start the helper at the beginning of the draft? What is someone starts the Tool after mid way in the draft?

**Comment by Manuel-777** [comment]
mtgatool should read the whole previous picks even if it is started mid-draft, this is to ensure you get all the data when you start mid game or mid draft.

That like you mention is to ensure the createDraft() function is called only at the beggining of it.

**Comment by Manuel-777** [comment]
yes, that is probably the case. What you are doing will create new draft on every draft pick, so its not optimal and may cause unexpected errors with createDraft(). A possible fix for your problem (having the log cleared up after the draft has started) is probably adding a check to see if the draft has started and making it positive on createDraft(), so that it doesnt get executed twice. which should then should be reset to zero on draft end.

**Comment by farazs** [comment]
what about passing draftId to createDraft and keeping track of draftIds so the same draft doesn't get created twice?

**Comment by Manuel-777** [comment]
I am trying to remove all the useless conditions from the draft log codes, today I removed some that allow the create draft method to execute again if it was not set before mid-draft.
[see here](https://github.com/Manuel-777/MTG-Arena-Tool/commit/ef987840691ed5e8e24b7ebbc5ddda0fb969997f)

I am yet to see how this will behave on very extreme situations (like missing a whole draft and then entering a new one in the middle?), But hopefully it will work well when closing MTGA in the middle of a draft, In my testing by replaying logs it works.

**Comment by farazs** [comment]
Sounds good. I didn't encounter any issues with the `createDraft` being called over and over. I was able to complete the draft with the condition removed but I will try the latest code next time I am drafting.

I noticed you also removed some console.log statements. I find those were helpful for me to debug (may be helpful for other people not familiar with the code as well). up to you if you want to remove though.
